sources/lib/api/gcp/bigquery/_dataset.py

```diff
@@ -251,7 +251,8 @@ def _retrieve_datasets(self, page_token, count):
                              info['datasetReference']['datasetId']), self._context)
                     for info in datasets]
       except KeyError:
-        raise Exception('Unexpected item list response.')
+        raise Exception('Unexpected item list response (missing datasetReference in %s)' %
```

Comment:
I am not sure how showing the JSON in the exception message helps -- it makes the error only more convoluted IMO... and not sure what the user can do differently to not get here.

Is this about improvement for our debugging purposes? If so maybe we need to actually have a debug mode, and in those cases write out more to stderr, rather than put data into exception messages.

Thoughts?